packages/task/src/browser/task-service.ts

```diff
@@ -173,6 +184,12 @@ export class TaskService implements TaskConfigurationClient {
             if (!this.isEventForThisClient(event.ctx)) {
                 return;
             }
+            if (!this.runningTasks.has(event.taskId)) {
+                this.runningTasks.set(event.taskId, { exitCode: new Deferred<number | undefined>(), terminateSignal: new Deferred<string | undefined>() });
+            }
+            this.runningTasks.get(event.taskId)!.exitCode.resolve(event.code);
+            this.runningTasks.get(event.taskId)!.terminateSignal.resolve(event.signal);
+            setTimeout(() => this.runningTasks.delete(event.taskId), 60 * 1000);
```

Comment:
You use `event.taskId` 6 times here, maybe it makes sense to use constant for `taskId`, but up to you.